services/autoscaling/src/simcore_service_autoscaling/modules/instrumentation/_utils.py

```diff
@@ -0,0 +1,43 @@
+import collections
+from collections.abc import Iterable
+from dataclasses import dataclass, field
+
+from aws_library.ec2._models import EC2InstanceData
+from prometheus_client import Gauge
+
+from ._constants import METRICS_NAMESPACE
+
+
+@dataclass
+class TrackedGauge:
```

Comment:
question out of curiosity: any reason why you went for putting `Gauge` as a member of the `TrackedGauge` class instead of using inheritance (TrackedGauge is childclass of Gauge)?

Reply:
I usually prefer to not go too much with inheritance but prefer composition as it is less likely to need changes when for example prometheus_client is updated. Also I would need to be compatible with whatever change in the constructors.

Some references:
https://www.digitalocean.com/community/tutorials/composition-vs-inheritance

Since I have absolutely no control on Gauge I did it this way. But I guess the inheritance would also work as it is not so different.

Reply:
thanks!